### DISCUSSION TITLE
How to add params to queryKey when using createQuery

### DISCUSSION THREAD

**Discussion by iamhaaamed** [TanStack/query#7734] [Q&A]
I have a query like this:
```
export const useRequest = createQuery<Response, Variables, AxiosError>({
  queryKey: ['requests',],
  fetcher: async ({ id }) => {
    const response = await client.get(`request-garage/${id}`);
    return response.data;
  },
});
```
I want to know how can I add params to the queryKey like this:
`queryKey: ['requests', **id**]`

**Comment by 87xie** [comment]
Create a new function to wrap the query.

```ts
export const useRequest = (id: string) => createQuery<Response, Variables, AxiosError>({
  queryKey: ['requests', id],
  queryFn() {
    const response = await client.get(`request-garage/${id}`);
    return response.data;
  },
});
```

**Reply by iamhaaamed** [reply]
> Create a new function to wrap the query.
> 
> ```ts
> export const useRequest = (id: string) => createQuery<Response, Variables, AxiosError>({
>   queryKey: ['requests', id],
>   queryFn() {
>     const response = await client.get(`request-garage/${id}`);
>     return response.data;
>   },
> });
> ```

i want to be able to pass other options to my hook:
const { data } = useRequest({ id: '1' }, { enabled: true, onSuccess:... });
